main.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2023 The Numaproj Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	sinksdk "github.com/numaproj/numaflow-go/pkg/sinker"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"go.uber.org/zap"
+)
+
+type awsSQSSink struct {
+	logger          *zap.SugaredLogger
+	sqsClient       *sqs.Client
+	queueName       string
+	region          string
+	awsAccessKey    string
+	awsAccessSecret string
+	awsBaseEndpoint string
+}
+
+// newAWSSQSSink will read the environment variable required to authenticate aws and publish data to SQS
+func newAWSSQSSink() *awsSQSSink {
+	logger := logging.NewLogger().Named("aws-sqs-sink")
+	queueName, ok := os.LookupEnv("AWS_SQS_QUEUE_NAME")
+	if !ok {
+		logger.Fatalln("AWS_SQS_QUEUE_NAME not found")
+	}
+	region, ok := os.LookupEnv("AWS_REGION")
+	if !ok {
+		logger.Fatalln("AWS_REGION not found")
+	}
+	accessKey, ok := os.LookupEnv("AWS_ACCESS_KEY")
```

Comment:
The credentials should not be mandatory, ppl might use things like [IAM Role for service account](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to access.

Reply:
+1